javascript/react/src/components/Codebook/CodebookFieldEditor.types.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { Field } from '@datashaper/schema'
+
+export enum CodebookFields {
+	DisplayName = 'display name',
+	Description = 'description',
+	DataType = 'data type',
+	DataNature = 'data nature',
+	Units = 'units',
+	Mapping = 'mapping',
+}
+
+export interface CodebookFieldEditorProps {
+	field: Field
+	onChange: (field: Field) => void
+	tableView?: boolean
+	hideLabel?: boolean
```

Comment:
Here is an idea for getting rid of the `hideLabel` prop to clean up the interface:

1. Add the styles props as noted above, and include one for `label`.
2. Do not use the label prop on individual controls, instead add a <Label> or even just a div that is styled with the `styles.label` prop.
3. On the table editor view, set the label.styles display to 'none'.

This would eliminate all of the custom label logic you have through a single style prop.